### PR TITLE
fix(core/utils): generate ids without diacritical marks

### LINKED ISSUE #1813: Autogenerated (section) IDs omit unicode characters
## Important info

* ReSpec version: 22.5
* [x] I did a "hard refresh", but it's still busted. 

## Description of problem

Autogenerated IDs of sections (and other parts of the document as well) based on headings omit unicode characters, creating nonsensical words in URLs.

I would expect the IDs to keep the words intact (i.e. do only lowercase, if that is necessary)

Optional, steps to reproduce:
1. Create a section with heading containing unicode characters, e.g. `Zkracování` and no `id` attribute
2. Observe that the generated id is e.g. `zkracov-n-` instead of `zkracování`

### PR DESCRIPTION
**Pull request by marcoscaceres**
* closes #1813

### CODE REVIEW

File: src/core/utils.js [hunk 1]
```diff
@@ -599,7 +599,10 @@ export function addId(elem, pfx = "", txt = "", noLC = false) {
   }
   let id = noLC ? txt : txt.toLowerCase();
   id = id
-    .replace(/[\W]+/gim, "-")
+    .trim()
+    .normalize("NFD")
+    .replace(/[\s]+/gim, "-")
```
Comment (sidvishnoi) [inline src/core/utils.js]: `\W+` works as we're expecting it to work.
Any reason to switch to `\s`?

**Comment by marcoscaceres** [comment]
Oh, I had issues with \W, but I guess maybe it was the ordering... dunno 🤷‍♂️ happy if the tests are passing

**Comment by marcoscaceres** [comment]
> Nice. Learnt about string.normalize today 

Yeah, me too. Did you read the Stack Overflow link from the original bug?

**Comment by sidvishnoi** [comment]
> Did you read the Stack Overflow link from the original bug?
Yes. And also a GitHub gist (Java based)

**Pull request merged**